tests/slsa_analyzer/ci_service/test_github_actions.py

```diff
@@ -1,87 +1,101 @@
+
+    # Test internal and reusable workflows.
+    # Parse GitHub Actions workflows.
+    root = GitHubNode(name="root", node_type=GHWorkflowType.NONE, source_path="", parsed_obj={}, caller_path="")
+    gh_cg = CallGraph(root, "")
+    workflow_path = os.path.join(resources_dir, "valid1.yaml")
+    parsed_obj = parse_action(workflow_path, macaron_path=MACARON_PATH)
+
+    callee = GitHubNode(
+        name=os.path.basename(workflow_path),
+        node_type=GHWorkflowType.INTERNAL,
+        source_path=workflow_path,
+        parsed_obj=parsed_obj,
+        caller_path="",
+    )
+    root.add_callee(callee)
+    github_actions.build_call_graph_from_node(callee)
+    assert [
+        "GitHubNode(valid1.yaml,GHWorkflowType.INTERNAL)",
+        "GitHubNode(apache/maven-gh-actions-shared/.github/workflows/maven-verify.yml@v2,GHWorkflowType.REUSABLE)",
+    ] == [str(node) for node in gh_cg.bfs()]
+
+    # Test internal and external workflows.
+    # Parse GitHub Actions workflows.
+    root = GitHubNode(name="root", node_type=GHWorkflowType.NONE, source_path="", parsed_obj={}, caller_path="")
+    gh_cg = CallGraph(root, "")
+    workflow_path = os.path.join(resources_dir, "valid2.yaml")
+    parsed_obj = parse_action(workflow_path, macaron_path=MACARON_PATH)
+
+    callee = GitHubNode(
+        name=os.path.basename(workflow_path),
+        node_type=GHWorkflowType.INTERNAL,
+        source_path=workflow_path,
+        parsed_obj=parsed_obj,
+        caller_path="",
+    )
+    root.add_callee(callee)
+    github_actions.build_call_graph_from_node(callee)
+    assert [
+        "GitHubNode(valid2.yaml,GHWorkflowType.INTERNAL)",
+        "GitHubNode(actions/checkout@v3,GHWorkflowType.EXTERNAL)",
+        "GitHubNode(actions/cache@v3,GHWorkflowType.EXTERNAL)",
+        "GitHubNode(actions/setup-java@v3,GHWorkflowType.EXTERNAL)",
+    ] == [str(node) for node in gh_cg.bfs()]
+
+
+def test_is_detected(github_actions: GitHubActions) -> None:
+    """Test detecting GitHub Action config files."""
+    assert github_actions.is_detected(str(ga_has_build_kws))
+    assert github_actions.is_detected(str(ga_no_build_kws))
+    assert not github_actions.is_detected(str(jenkins_build))
+
+
+def test_get_workflows(github_actions: GitHubActions) -> None:
+    """Test detection of reachable GitHub Actions workflows."""
+    # Test GitHub Actions workflows that contain build commands.
+    expect = [str(path) for path in ga_has_build_kws.joinpath(".github", "workflows").glob("*")]
+    expect.sort()
+    workflows = github_actions.get_workflows(str(ga_has_build_kws))
+    workflows.sort()
+    assert workflows == expect
+
+    # Test GitHub Actions workflows that do not contain build commands.
+    expect = [str(path) for path in ga_no_build_kws.joinpath(".github", "workflows").glob("*")]
+    expect.sort()
+    workflows = github_actions.get_workflows(str(ga_no_build_kws))
+    workflows.sort()
+    assert workflows == expect
```

Comment:
This could also be
```suggestion
    expect = sorted([str(path) for path in ga_no_build_kws.joinpath(".github", "workflows").glob("*")])
    assert sorted(github_actions.get_workflows(str(ga_no_build_kws)))
    assert workflows == expect
```
which is arguably a bit closer to the original, but I also don't see any problem with using a separate call to `.sort()`.

Reply:
See commit [416179a](https://github.com/oracle/macaron/pull/591/commits/416179a5d4503986e9cee8179b7e3917909a2080)